core/py/src/braintrust_core/util.py

```diff
@@ -12,6 +12,13 @@ def as_json(self, **kwargs):
         """Serialize the object to JSON."""
         return json.dumps(self.as_dict(), **kwargs)
 
+    @classmethod
+    def from_dict(cls, d: Dict):
+        """Deserialize the object from a dictionary."""
+        fields = set(f.name for f in dataclasses.fields(cls))
+        filtered = {k: v for k, v in d.items() if k in fields}
```

Comment:
I'm guessing this doesn't work for nested dataclasses? I think it won't even crash at runtime because it's just the types that are different. Perhaps we could:
- assert `type(v) == field.type`
- optionally, recursively invoke `from_dict` if `isinstance(field.type, SerializableDataClass)`

Reply:
I'll just add a comment that it's shallow

Reply:
Seems harmless to also assert that `isinstance(type(v), field.type)`. When would someone want to invoke this with a type that's not the same as the declared dataclass type.

Reply:
ok will add

Reply:
Unfortunately `type` doesn't really seem to be what we want:


```
In [4]: dataclasses.fields(EvalCase)[0]
Out[4]: Field(name='input',type=~Input,default=<dataclasses._MISSING_TYPE object at 0x102c87cd0>,default_factory=<dataclasses._MISSING_TYPE object at 0x102c87cd0>,init=True,repr=True,hash=None,compare=True,metadata=mappingproxy({}),kw_only=False,_field_type=_FIELD)

In [5]: dataclasses.fields(EvalCase)[0].type
Out[5]: ~Input

In [6]: isinstance(None, dataclasses.fields(EvalCase)[0].type)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 1
----> 1 isinstance(None, dataclasses.fields(EvalCase)[0].type)

TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
```